moon/controller.py

```diff
@@ -328,14 +320,17 @@ def try_step_around(self):
         self.go_straight(3.0, timeout=timedelta(seconds=20))
         self.turn(math.radians(-90), timeout=timedelta(seconds=10))
 
-    def run(self):
+    def wait_for_init(self):
         try:
             print('Wait for definition of last_position and yaw')
-            while self.last_position is None or self.yaw is None:
-                self.update()  # define self.sim_time
+            while self.sim_time is None or self.last_position is None or self.yaw is None:
+                self.update()  # define self.time
             print('done at', self.sim_time)
+        except BusShutdownException:
+            pass
 
-
+    def run(self):
+        try:
```

Comment:
`wait_for_init()` is not called here - is it expected?

Reply:
yes, controller.py is now meant to be an abstract class in that all subclasses are expected to implement their own run() (as they do) and put wait_for_init in it (har) as its first step.  It's because there were a few scenarios where the robot was happy to use the baseline driving except it needed to do one thing (e.g. fetch origin, list of volatiles, etc) before getting started which in turn needed an initialized state (that is achieved in wait_for_init). Hence, the implementation would be something like:

```
def run(self):
  self.wait_for_init()
  do_specific_operation()
  super().run()
```

However, as development progressed, the importance of this approach receded and as custom driving routines are developed for each robot, the baseline driving can be retired.